app/Models/Track.php

```diff
@@ -688,7 +699,7 @@ public function getFilenameFor($format)
 
         $format = self::$Formats[$format];
 
-        return "{$this->id}.{$format['extension']}";
+        return "{$this->id}-v{$this->current_version}.{$format['extension']}";
```

Comment:
We have a leaky abstraction here and on line 729 - the filename on the disk should be an implementation detail of the `TrackFile` class and its generation is probably being unnecessarily duplicated in this class. Consider refactoring this and `getDownloadFilenameFor`.